UnityProject/Assets/Scripts/Managers/SoundManager.cs

```diff
@@ -256,42 +207,73 @@ private string[] GetMatchingSounds(string pattern)
+	/// </summary>
+	/// <param name="addressableAudioSources">The sound to be played.</param>
+	/// <param name="worldPos">The position at which the sound is played</param>
+	/// <param name="polyphonic">Is the sound to be played polyphonic</param>
+	/// <param name="audioSourceParameters">Extra parameters of the audio source.</param>
+	/// <param name="Global">Does everyone will receive the sound our just nearby players</param>
+	/// <param name="sourceObj">The object that is the source of the sound</param>
+	/// <param name="shakeParameters">Camera shake effect associated with this sound</param>
+	/// <returns>The SoundSpawn Token generated that identifies the same sound spawn instance across server and clients</returns>
+	public static Task<string> PlayNetworkedAtPos(AddressableAudioSource addressableAudioSource, Vector3 worldPos, AudioSourceParameters audioSourceParameters,
+		bool polyphonic = false, bool Global = true, GameObject sourceObj = null, ShakeParameters shakeParameters = null)
+	{
+		return PlayNetworkedAtPos(new List<AddressableAudioSource> { addressableAudioSource }, worldPos, audioSourceParameters, polyphonic, Global, sourceObj, shakeParameters);
 	}
 
 	/// <summary>
 	/// Serverside: Play sound at given position for all clients.
-	/// Accepts "#" wildcards for sound variations. (Example: "Punch#")
+	/// If more than one sound is specified, the sound will be chosen at random
 	/// </summary>
-	/// <param name="sndName">The name of the sound to be played</param>
+	/// <param name="addressableAudioSources">The sound to be played.  If more than one is specified, a single one will be picked at random</param>
 	/// <param name="worldPos">The position at which the sound is played</param>
 	/// <param name="polyphonic">Is the sound to be played polyphonic</param>
 	/// <param name="audioSourceParameters">Extra parameters of the audio source.</param>
 	/// <param name="Global">Does everyone will receive the sound our just nearby players</param>
 	/// <param name="sourceObj">The object that is the source of the sound</param>
 	/// <param name="shakeParameters">Camera shake effect associated with this sound</param>
-	public static void PlayNetworkedAtPos(string sndName, Vector3 worldPos, AudioSourceParameters audioSourceParameters, bool polyphonic = false, bool Global = true, GameObject sourceObj = null, ShakeParameters shakeParameters = null)
+	/// <returns>The SoundSpawn Token generated that identifies the same sound spawn instance across server and clients</returns>
+	public static async Task<string> PlayNetworkedAtPos(List<AddressableAudioSource> addressableAudioSources, Vector3 worldPos, AudioSourceParameters audioSourceParameters,
+			bool polyphonic = false, bool Global = true, GameObject sourceObj = null, ShakeParameters shakeParameters = null)
 	{
-		sndName = Instance.ResolveSoundPattern(sndName);
+		AddressableAudioSource addressableAudioSource = await GetAddressableAudioSourceFromCache(addressableAudioSources);
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)